sycl/doc/extensions/proposed/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -116,7 +130,7 @@ A pipe type is a specialization of the pipe class:
 ----
 template <typename name,
           typename dataT,
-          size_t min_capacity = 0>
+          typename propertiesT = properties<>>
```

Comment:
Using properties here is nice, but this is an API-breaking change to a supported extension.  Existing code using the old API will no longer compiler, right?

Reply:
A common strategy in sycl runtime is to keep the origin pipe class, and define a new pipe class definition that uses new property list. Do we want to use this strategy here, or just delete the old class?

If keep old class:
Pro: no ABI break
Con: tech debt

Reply:
This is exactly the difference between an "experimental" extension vs. a "supported" one.  If it is "supported", we promise not to break API compatibility with existing code without going through a deprecation process.

If the FPGA team wants the freedom to change APIs without deprecation, we should consider making this an experimental API (which includes moving it to the the `experimental` namespace).

Reply:
Indeed this breaks the ABI, and I don't see any trick using [SFINAE](https://en.cppreference.com/w/cpp/language/sfinae) via [`std::enable_if`](https://en.cppreference.com/w/cpp/types/enable_if) to retain compatibility.

@GarveyJoe, am I missing anything obvious?

 @gmlueck, would you accept a global macro to opt `pipe` into the properties parameter, along with a deprecation notice for the `_min_capacity` parameter when the macro is undefined? The properties parameter would become the default at some later point, after compile-time constant properties have been moved out of experimental.

Reply:
@pcolberg @GarveyJoe @gmlueck Is there anything we still need to resolve in this conversation?